### DISCUSSION TITLE
trigger event inside  hits

### DISCUSSION THREAD

**Discussion by maxpanakov** [algolia/instantsearch#5178] [Q&A]
Hey, i can't find how to set event on buttons. i tried with "Insights", but im not using algolia. and im not sure that it's the right way to handle events. For example buttons "Add to cart" or "Add to compare"
![image](https://user-images.githubusercontent.com/9987863/206862295-3b37c850-83f2-4cba-b492-1e78e38e68cf.png)

**Comment by maxpanakov** [comment]
![image](https://user-images.githubusercontent.com/9987863/206862652-caa1efb2-7b55-447e-923f-5838d7f9f5c5.png)
I tried to implement insight, but it asking for search, but i have this error 
![image](https://user-images.githubusercontent.com/9987863/206862711-60586b3e-ce4b-40a2-aa9f-f70f07b4309b.png)

**Comment by Haroenv** [comment]
you can add `onClick` to the elements:

```js
  instantsearch.widgets.hits({
    container: '#hits',
    templates: {
      item: (hit, { html, components }) => html`
        <div>
          <img src="${hit.image}" align="left" alt="${hit.name}" />
          <div class="hit-name">
            ${components.Highlight({ attribute: 'name', hit })}
          </div>
          <button type="button" onClick="${(e) => console.log(e, hit)}">click</button>
        </div>
      `,
    },
  })
```

https://codesandbox.io/s/eager-cannon-j3todq?file=/src/app.js
